File: packages/typescript/src/client/Cache.ts
import type { Client } from "../clients/Client.ts";

export namespace Cache {
  export type ClearProps = Record<string, unknown>;
}

export class Cache {
  private client: Client;

  constructor(client: Client) {
    this.client = client;
  }

  save(): Promise<void> {
    return this.client.saveCache();
  }

  discard(): Promise<void> {
    return this.client.discardCache();
  }
}

File: packages/java/src/main/java/ai/alumnium/Cache.java
package ai.alumnium;

import ai.alumnium.client.HttpClient;

public class Cache {
  private final HttpClient client;

  public Cache(HttpClient client) {
    this.client = client;
  }

  public void save() {
    client.saveCache();
  }

  public void discard() {
    client.discardCache();
  }
}
